docs/items/overrides.md

```diff
@@ -0,0 +1,46 @@
+Item Property Overrides
+=======================
+
+Items do not have something analogous to blockstate JSONs, where you can define models in `models/block/` and then stitch them into one big model in a blockstate JSON under `blockstates/` depending on variants. Instead, you must assign an item a single model defined under `models/item/`. Then, the item will expose certain properties from the code. These properties are defined as a name string and a function that maps `ItemStack`s to arbitrary float values. The model JSON defines the default model for the item, and has an extra `"overrides"` tag, which is a list of predicates and a model to use if the predicate is true. The format can be seen on the [wiki][].
```

Comment:
First of all, yes they do, like you explained. Blockstate jsons can be used for items.
Then, "stitch them into one big model" is very inaccurate. The blockstate json is not itself a model, it simply points to models and describes how they should be used (transformations, textures, etc.).
I would not describe property overrides as the alternative to blockstates, they are simply a different concept.